crates/flashblocks/p2p/src/events/listerners.rs

```diff
@@ -0,0 +1,28 @@
+use async_trait::async_trait;
+use reth_network::events::PeerEvent;
+
+use crate::events::{NetworkPeersContext, PeerEventsListener};
+
+/// Logs error when a trusted peer disconnects.
+pub struct TrustedPeerDisconnectedAlert;
+
+#[async_trait]
+impl<N> PeerEventsListener<N> for TrustedPeerDisconnectedAlert
+where
+    N: NetworkPeersContext + Send + Sync,
+{
+    async fn on_peer_event(&self, _network: &N, event: &PeerEvent) {
```

Comment:
Maybe I missed it, but where are we checking that these peers are trusted, and not regular peers?